contracts/events/TopicEvent.sol

```diff
@@ -299,54 +292,43 @@ contract TopicEvent is ITopicEvent, BaseContract, Ownable {
+    * @notice Calculates the BOT and QTUM tokens won based on the sender's contributions.
+    * @return The amount of BOT and QTUM tokens won.
     */
-    function calculateQtumContributorWinnings() 
+    function calculateWinnings()
         public 
         view
         inCollectionStatus()
-        returns (uint256)  
-    {
-        uint256 senderContribution = balances[resultIndex].bets[msg.sender];
-        uint256 winnersTotal = balances[resultIndex].totalBets;
-        uint256 losersTotalMinusCut = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
-            if (i != resultIndex) {
-                losersTotalMinusCut = losersTotalMinusCut.add(balances[i].totalBets);
-            }
-        }
-        losersTotalMinusCut = losersTotalMinusCut.mul(90).div(100);
-        uint256 senderContributionMinusCut = senderContribution.mul(90).div(100);
-        return senderContribution.mul(losersTotalMinusCut).div(winnersTotal).add(senderContributionMinusCut);
-    }
-
-    /*
-    * @notice Calculates the QTUM and BOT tokens won based on the sender's BOT contributions.
-    * @return The amount of QTUM and BOT tokens won.
-    */
-    function calculateBotContributorWinnings()
-        public
-        view
-        inCollectionStatus()
-        returns (uint256, uint256) 
+        returns (uint256, uint256)  
     {
-        // Calculate BOT won
-        uint256 senderContribution = balances[resultIndex].votes[msg.sender];
+        // Calculate BOT winnings
+        uint256 contribution = balances[resultIndex].votes[msg.sender];
         uint256 winnersTotal = balances[resultIndex].totalVotes;
         uint256 losersTotal = 0;
         for (uint8 i = 0; i < numOfResults; i++) {
             if (i != resultIndex) {
                 losersTotal = losersTotal.add(balances[i].totalVotes);
             }
         }
-        uint256 botWon = senderContribution.mul(losersTotal).div(winnersTotal).add(senderContribution);
-
-        // Calculate QTUM won
-        uint256 qtumReward = totalQtumValue.mul(10).div(100);
-        uint256 qtumWon = senderContribution.mul(qtumReward).div(winnersTotal);
+        uint256 botWon = contribution.mul(losersTotal).div(winnersTotal).add(contribution);
+        uint256 qtumReward = totalQtumValue.mul(QTUM_PERCENTAGE).div(100);
+        uint256 qtumWon = contribution.mul(qtumReward).div(winnersTotal);
+
+        // Calculate QTUM winnings
+        contribution = balances[resultIndex].bets[msg.sender];
+        winnersTotal = balances[resultIndex].totalBets;
+        losersTotal = 0;
+        for (i = 0; i < numOfResults; i++) {
+            if (i != resultIndex) {
+                losersTotal = losersTotal.add(balances[i].totalBets);
+            }
+        }
+        uint256 percentAfterCut = uint256(100).sub(uint256(QTUM_PERCENTAGE));
+        losersTotal = losersTotal.mul(percentAfterCut).div(100);
+        uint256 contributionMinusCut = contribution.mul(percentAfterCut).div(100);
```

Comment:
Wait a second, the 10% is from all bets including winners? If I win I need to spare 10% of my QTUM contribution to Voters? Then it's likely that I win but my winnings is less than contribution, in the event that losersTotal is tiny.

Reply:
What about only extract those 10% from losers?

Reply:
Ya that was how TT wanted it. I think the percentage is not set in stone at this point. He keeps changing the number. I believe last time he mentioned it was like 1%.

The only case where the winnings is less than the contribution is if only a single user bet, and the COracle sets the result on that same index. Then the percentage cut gets removed from the user and it goes towards the COracle.

In other cases where there are multiple bets on different results. The QTUM winner should always get their original contribution back (minus the cut) + percentage of losers QTUM (minus the cut).

Reply:
Yeah, let me run by him. The fact that I can lose my qtum even when I'm winning will stop me from trying out an app. When people considering an investment, the bottom line is not lose money.

Reply:
Double checked with Xiahong. Yeah the 10% is from losers ...

Reply:
gotcha, ok will make changes accordingly